cmd/utils.go

```diff
@@ -12,3 +12,25 @@ func fileExists(filename string) bool {
 	}
 	return !info.IsDir()
 }
+
+// prepare golang private packages for whitesource and blackduck(detectExecuteScan)
+func prepareGolangPrivatePackages(privateModules, privateModulesGitToken string) error {
```

Comment:
since this a common functionality used for all golang related steps, `utils.go` does not seem like the right place to add this function

i would rather create a new `pkg : pkg/golang/golang.go` and move this functionality there . since it will more readable and all steps should refer to this package

Reply:
Good point. But if  I take it out the 'package cmd' then I need to changes in golangBuild step. (struct golangBuildOptions, func newGolangBuildUtils, func  readGoModFile). Will it be Ok?

Reply:
you can keep it `cmd ` package for the `golangBuild `step and make the `whitesourceExecuteScan `and `detectExecutescan` use the new package . 

since that is a side effect there will some refactoring needed, i will create a seperate PR for the `golangBuild `and get rid of the coce in the `cmd `package then

Reply:
I moved it and tested it through the Jenkins pipeline - success